### ISSUE TITLE
Game board and Game00 prototype

### ISSUE THREAD

**Issue by interrogator** [acdh-oeaw/wugsy#10]
Currently, the backend gives dummy data to the frontend on request. This data should be able to be turned into a visual game, with a ruleset provided by #1.

Doing this will (probably) require building a game base class in `src/static/site/js/game.js`, which all game types inherit from. Then, in the same or a different file, build a Game00 subclass that follows the rules for this game.

**Comment by interrogator** [comment]
Want to join in on this one, @notsomes ?

**Comment by interrogator** [comment]
![mockup](https://user-images.githubusercontent.com/10471132/35096199-9be8d7fe-fc4b-11e7-9dca-ddb980fc54ca.png)

Basic design of base game, irrespective of game rules ... degisn by @TPalfinger

**Comment by interrogator** [comment]
@notsomes do you have any questions on this before you begin? Would you like more permissions with regad to this repo, for example.

**Comment by notsomes** [comment]
I've finally been able to start looking at the code. I hope to have a clear understanding of it asap (I'm also checking how you have structured the how project, how Docker works, etc etc).

Once I get it I'll try to do some quick things just to know how everything should work and then we can move on to develop the actual game(s). Will keep you updated.

**Comment by interrogator** [comment]
Yes, the priority is for you to have a working environment, preferably using Docker. Once you can serve and visit the web app, go to `Games` and generate one. You'll see the JSON on screen, and in the console. It's dummy data now, but the format is the same as what the real data will be.

**Comment by notsomes** [comment]
I have a few questions regarding this game and its set of rules, or at least what some things are meant to represent or what do they mean, so I'd like both of you @TPalfinger and @interrogator –maybe even @amelieacdh can contribute– (if you have anything to say about any point, as you see fit) to clarify them if possible:

1. In this game's rules it is expected to play multiple rounds. The rounds work in the same way (in terms of mechanics – linking words between them), though it is not clear to me what separates a "primary" link from a "secondary" link from a "tertiary" link.

2. Are "primary/secondary/tertiary/..." links defined by the user and how he sees the words/concepts/relations? For example, one person may see apple-fruit as a primary link, and apple-pear as a secondary link, while other one may see it the other way around and create those links in different rounds.

3. Are "primary/secondary/tertiary/..." links expected to be defined by ourselves by giving each level some kind of description/category? For example, we may label each round with some word or category we think goes well with the words appearing in the cards, providing some context for the round. Example: _"Round One! Create links between the words in the context of "Festivities"!"_ – User proceeds to link grapes with Christmas, turkey with Thanksgiving, etc etc until the timer marks zero. _"Round Two!" Create links between the words in the context of "bla bla bla"_ – Repeat process. Would this work better (in terms of structuring the game in some way)?

4. While the timer seems to be the logical approach to limit rounds, it may not work. What if someone is able (somehow) to create all of the links he can think of during the time the first rounds lasts? What is he supposed to do in the next rounds? Please detail the game mechanics in more depth if possible.

5. Regarding points, I'd go with something simple at first, such as awarding points 5 times the number of links created or something silly like that. Once we have a good database of links provided by users, though, I'd change the approach and award points (or something not points) or let's call it a "similarity score" in terms of how the user created links in relation to other people. We discussed this already and it seemed like a good/fun idea to play around with. Thoughts?

6. Doesn't look clear to me what the text field function is. I mean, I see you want users to "describe" the concept, though I think concepts should be crystal clear (the more the better). I already mentioned concepts such as "Festivities", but we may even go further and make them even more concrete. For example, at the time of creating the game we may have some initial categories (e.g. Festivities) and then concepts within those categories (Christmas, Thanksgiving,...), each of the latter making for a round. This may change the whole "primary/secondary/tertiary level" approach but could be explored as another developing/game option.

Please provide answers or thoughts about every of those questions, and of course don't take it as a critique but rather a starting point to build Game00 or other multiple games! :D

**Comment by TPalfinger** [comment]
@notsomes (not sure if you get a notification without that) 

1. See description in 2. short: primary links are links of the first round and between the given concept on the concept cards linked to the text_cards. 
2. Primary/secondary etc. Links are defined by the rounds. In round 1 player can only link the given concept on the concept_card (e.g. marriage/other festivity) to the text_cards on the field (marriage with cake; marriage with apple etc.). In round 2. players can pick of the beforehand selected text_cards and link them to other text_cards (this is not possible in round one); in the second round players are no longer allowed to link between the concept_card and the text_cards. In the second round players link between e.g from cake to flour (there is no direct link between marriage and flour, but a secondary link over cake). This means primary link is always the link(s) between our concept and the selected text_fields, while secondary links are between selected text_fields and other text fields. So we might draw something like "mind trees"
3. The links are "defined" by the rounds they are played; In the first round players get a concept on the concept_card where the whole linking process starts from. So yes, people will make links between this starting point (no matter whats on the card, can be marriages, funerals etc.) and the text_fields, and over this concept we moderate implicitly the process. 
4. It is not possible for the player to make secondary links in the first round. In the first round he is only able to make links from the concept to the text_fields. But we might need a finished button or something (and a skip button if he is not able to find any link), if the player is faster than the time.
5. in the long run we will need a more complex score system. But we might discuss this issue (as it gets complicated very fast) in a separated meeting (or discussion here :)). For the beginning I think it is enough to have a simple but justifiable score system running --> but I do not know if we need place holder in the game for a more complex system 
6. The concepts and the text fields may be clear, but why players make these links not (or why they are not similar links made by other players or some sample data etc.)

Hope things are clearer know :) And do not hesitate to input your thoughts!

**Comment by ameliedorn** [comment]
I think we @TPalfinger @notsomes and me should briefly exchange on the game details in a separate meeting; also we will decide what part of our data can serve as an initial data set to use.

**Comment by ameliedorn** [comment]
@notsomes @interrogator @TPalfinger what do we need to tackle next?

**Comment by interrogator** [comment]
I'm going to make an issue for the newest game, which we fleshed out in Salamanca. Work on the card games is probably best to stop right now.
